src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonEncodingStreamWrapper.cs

```diff
@@ -174,15 +174,18 @@ public override void Flush()
             _stream.Flush();
         }
 
-        public override int Read(byte[] buffer, int offset, int count)
+        public override int Read(byte[] buffer, int offset, int count) =>
+            Read(new Span<byte>(buffer, offset, count));
+
+        public override int Read(Span<byte> buffer)
         {
             try
             {
                 if (_byteCount == 0)
                 {
                     if (_encodingCode == SupportedEncoding.UTF8)
                     {
-                        return _stream.Read(buffer, offset, count);
+                        return _stream.Read(buffer);
```

Comment:
Note that `_stream` is always a `BufferedStream`, which also overrides `Read(Span<byte>)`, so we shouldn't have the same problem here.